dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -114,24 +114,26 @@ def refute_estimate(self):
 
                 if callable(action):
                     new_outcome = action(X, **func_args)
+
                 elif action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
-                    if save_estimators:
-                        estimator = self._estimate_dummy_outcome(func_args, action, new_outcome)
-                        saved_estimator_dict[action + str(transform_num)] = estimator
-                    
                     if action + str(transform_num) in saved_estimator_dict:
                         estimator = saved_estimator_dict[action + str(transform_num)]
                         new_outcome = estimator(X)
                     else:
                         estimator = self._estimate_dummy_outcome(func_args, action, new_outcome)
                         new_outcome = estimator(X)
+                        if save_estimators:
+                            estimator = self._estimate_dummy_outcome(func_args, action, new_outcome)

```

Comment:
you don't need this line 126 since line 123 already does it. Why estimate twice?

Reply:
It only estimates the value if the relationship between the estimator and the covariates has not been computed. If the value has been estimated, it run the ``if(condition) ...```, otherwise, it runs the ```else:...```. In the else, block, it also checks if the estimator is eligible for saving; if so, it saves the value within the dictionary.

Reply:
Yeah, that is unnecessary. Will remove it in the upcoming commit.